repositories/src/main/scala/cromwell/pipeline/datastorage/dto/ProjectConfiguration.scala

```diff
@@ -6,16 +6,24 @@ import ProjectFile.pathFormat
 import org.mongodb.scala.bson.collection.immutable.Document
 import play.api.libs.json.{ Json, OFormat }
 
-case class ProjectFileConfiguration(path: Path, inputs: List[FileParameter])
+case class ProjectFileConfiguration(path: Path, inputs: Set[FileParameter])
 
 object ProjectFileConfiguration {
   implicit val projectFileConfigurationFormat: OFormat[ProjectFileConfiguration] = Json.format
+
+  def toMap(config: ProjectFileConfiguration): Map[Path, Map[String, TypedValue]] =
+    Map(
+      config.path -> config.inputs.flatMap(fileParameter => Map(fileParameter.name -> fileParameter.typedValue)).toMap
+    )
+
+  def fromMap(path: Path, inputs: Map[String, TypedValue]): ProjectFileConfiguration =
```

Comment:
`fromMap` name looks misleading. We are creating `FileParameter` from a map, but `ProjectFileConfiguration` is created using two regular parameters.
If we have methods like `toSmth` / `fromSmth` then I would say the following code should work
```
val config1: ProjectFileConfiguration = ???
val config2: ProjectFileConfiguration = fromMap(toMap(config1))
```
But your implementation won't compile which is extremely confusing.

Reply:
Renamed and moved these methods from here.